src/Simplex/Chat/Store/Direct.hs

```diff
@@ -267,6 +274,34 @@ deleteContactProfile_ db userId contactId =
     |]
     (userId, contactId)
 
+deleteUnusedProfile_ :: DB.Connection -> UserId -> ProfileId -> IO ()
+deleteUnusedProfile_ db userId profileId =
+  DB.executeNamed
+    db
+    [sql|
+      DELETE FROM contact_profiles
+      WHERE user_id = :user_id AND contact_profile_id = :profile_id
+        AND 1 NOT IN (
+          SELECT 1 FROM connections
```

Comment:
I think connections don't require checking for regular (non incognito) profiles. Is it correct that this function is only used for such profiles?